src/components/GameBox/GameBox.js

```diff
@@ -0,0 +1,471 @@
+      if (allUtilities.length === 2)
+        return 10 * diceValues.one * diceValues.two;
+    }
+  };
+
+  const rentProperty = () => {
+    const currentOwner = getPropertyOwner();
+
+    const currentRent = getCurrentRentForProperty();
+    currentPlayer.balance -= currentRent;
+    currentOwner.balance += currentRent;
+
+    const newPropertyData = currentOwner.ownedProperties.map((property) => {
+      if (property.name === name)
+        return {
+          ...property,
+          rentLevel: property.rentLevel + 1 > 5 ? 5 : property.rentLevel + 1,
+        };
+      else return property;
+    });
+
+    currentOwner.ownedProperties = newPropertyData;
+    const message = `${currentPlayer.name} rented ${name} at $${currentRent} from ${currentOwner.name}`;
+    showToast(message);
+    transactionLog(message);
+    setPlayerAction(null);
+    toggleCurrentTurn();
+  };
+
+  const buyProperty = () => {
+    currentPlayer.ownedProperties.push({
+      name,
+      rentLevel: 0,
+      timeStamp: Date.now(),
+      type: boxType.type,
+    });
+    currentPlayer.balance -= boxType.price;
+    currentPlayer.playerTurn += 1;
+    const message = `${currentPlayer.name} bought ${name} at $ ${boxType.price}`;
+    showToast(message);
+    transactionLog(message);
+    setPlayerAction(null);
+    toggleCurrentTurn();
+  };
+
+  const handlePropertyTransaction = () => {
+    if (playerAction === "Buy") {
+      buyProperty();
+    } else if (playerAction === "Rent") {
+      rentProperty();
+    }
+  };
+
+  const handlePropertyDealing = () => {
+    const isPropertyOwnedByCurrentPlayer = currentPlayer.ownedProperties.find(
+      (property) => property.name === name
+    );
+    if (isPropertyOwnedByCurrentPlayer) {
+      toggleCurrentTurn();
+      return;
+    }
+    if (!isPropertyOwnedByCurrentPlayer) {
+      const currentOwner = getPropertyOwner();
+      if (currentOwner) {
+        setPlayerAction("Rent");
+      } else {
+        setPlayerAction("Buy");
+      }
+    }
+  };
+
+  const showPlayerCurrentPosition = () => {
+    const colors = [...players]
+      .filter((player) => player.currentIndex === id)
+      .map((player) => player.color);
+
+    return colors.map((color) => (
+      <div style={{ backgroundColor: color }} className="player-box" />
+    ));
+  };
+
+  const getPropertyBoughtColor = () => {
+    const currentOwner = getPropertyOwner();
+    if (currentOwner) {
+      return (
+        <div
+          className="owner"
+          style={{ backgroundColor: currentOwner.color }}
+        ></div>
+      );
+    } else return null;
+  };
+
+  const getRandomElementFromArray = (array) => {
+    return array[Math.floor(Math.random() * array.length)];
+  };
+
+  const chanceAction = (action) => {
+    const message = `${currentPlayer.name} - ${action}`;
+    showToast(message);
+    chanceLog(message);
+
+    if (
+      action ===
+      "GET OUT OF JAIL FREE. This card may be kept until needed or traded."
```

Comment:
`switch` case is a better option here than using the `if` condition.
Keeping these big strings as actions results in poor code readability.
I would suggest to keep them in uppercase constant variables like `GET_OUT_JAIL_CARD` or something similar.